rerun_cpp/src/rerun/demo_utils.hpp

```diff
@@ -30,7 +30,7 @@ namespace rerun {
         inline std::vector<T> linspace(T start, T end, size_t num) {
             std::vector<T> linspaced(num);
             std::generate(linspaced.begin(), linspaced.end(), [&, i = 0]() mutable {
-                return start + static_cast<T>(i++) * (end - start) / static_cast<T>(num - 1);
+                return static_cast<T>(start + static_cast<T>(i++) * (end - start) / static_cast<T>(num - 1));
```

Comment:
this makes no sense to me - isn't this casting `T` to `T`? Are you sure this is needed?

Reply:
Yeah it's this weird quirk of C++ where `char + char == int`

Reply:
(you know _that_ weird quirk. Not like the others)
Bites me every time!

https://godbolt.org/z/4W8xYnqKq

actually only works with unsigned char, which is also the case I got a warning in (uint8_t)

Reply:
God damn you C++